### DISCUSSION TITLE
How to load relationship in TextInput Form builder?

### DISCUSSION THREAD

**Discussion by alisalehi1380** [filamentphp/filament#13259] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.0

### How can we help you?

![image](https://github.com/filamentphp/filament/assets/111766206/91c95b2a-1d3c-4226-86a4-e682b18bfcfa)

I have f_name & l_name in users table and I wanna show and able user to `profile/1/edit page`, editing them.

each row of profiles `BelongsTo` users table.

I wrote this code but not working as well.

**Comment by alisalehi1380** [comment]
![image](https://github.com/filamentphp/filament/assets/111766206/df603aeb-374d-4d44-ae25-6d74b4bce40e)

I resolved my issue by overwriting `fillForm` method in `EditProfile` class but I guess there exists a better way to handle it .

**Comment by ifteakhor** [comment]
Check this page, maybe you will find a solution for it 

https://filamentphp.com/docs/3.x/forms/advanced#saving-data-to-relationships

**Reply by leandrocfe** [reply]
Shouldn't it be a Select with relationship?
https://filamentphp.com/docs/3.x/forms/fields/select#integrating-with-an-eloquent-relationship

```php
Select::make('user_id')
    ->relationship(name: 'user', titleAttribute: 'name')
```

**Reply by alisalehi1380** [reply]
Yeah @leandrocfe, but each user just has one profile!

**Reply by alisalehi1380** [reply]
> Check this page, maybe you will find a solution for it
> 
> https://filamentphp.com/docs/3.x/forms/advanced#saving-data-to-relationships

tnks @SaidkhojaIftikhor ✅✌️
This worked but is there another way without adding Fieldset?

**Reply by leandrocfe** [reply, marked as answer]
Do you want to do this?

```php
Group::make([
        TextInput::make('name')
    ])
->relationship('user')
```

**Reply by alisalehi1380** [reply]
exactly! Thanks a billion.🙌
I knew filament was better can handling it. 😅

**Reply by wallisonfelipe** [reply]
Nice!!!
